[material-ui][AvatarGroup] Add `renderSurplus` prop (#39283)
mui/material-ui: docs/data/material/components/avatars/avatars.md

{{"demo": "TotalAvatars.js"}}

### Custom surplus

> Set the `renderSurplus` prop as a callback to customize the surplus avatar. The callback will receive the surplus number as an argument based on the children and the `max` prop.

Reviewer: ```suggestion
Set the `renderSurplus` prop as a callback to customize the surplus avatar. The callback will receive the surplus number as an argument based on the children and the `max` prop, and should return a `React.ReactNode`.
```

> Set `renderSurplus` prop as a callback to customize the surplus avatar. The callback will receive one argument which is the number of surpluses based on the children and `max` prop.

Reviewer: ```suggestion
Set the `renderSurplus` prop as a callback to customize the surplus avatar. The callback will receive the surplus number as an argument based on the children and the `max` prop.
```

> This is useful when you need to render the surplus based on the data sent from the server.

Reviewer: ```suggestion
The `renderSurplus` prop is useful when you need to render the surplus based on the data sent from the server.
```
